### PR TITLE
Fixes #1707 Gageplots Feedback - Annual Peak Streamflow plot

### LINKED ISSUE #1707: Gageplots Feedback - Annual Peak Streamflow plot
More feedback may be added later! This is what we have received so far.

- [x] Consider adding option for a log scale
- [x] It should be noted that the % AEP flows are the at-site stats as opposed to the weighted %AEP flows (weighting of at-site values with regression values)
     - Maybe add a dropdown for data sources for each plot -- this way users can see exactly what stats/data is being used (see #1709)
     - (Awaiting guidance from Pete M on the wording of the note)
- [x] "It’s confusing that the x-scale shown for the peak flows is only correct for the default window"
     - [x] We're going to try: if the x-scale is zoomed to be greater than 1 year, plot the points on their actual year & uncheck the "Plot Annual Streamflow Peaks on Latest Year" box  
     - [x] Consider allowing an option to plot the peak flows on any user-specified year: drop-down options would be to plot points on actual date, 2023, 2022, 2021, etc... 
- When you use the slider at the bottom of the graph, all the peak flows get pushed to the far right. It would be better if only the peak flows associate with the period chosen on the slider would show up and use the x-scale being used for the daily mean streamflow.
     - See previous point 
- [x] For some sites (06334330 for example) the annual peak streamflow plot loads in blank and only by zooming out the slider can one see the peak flow values. Looks like the peaks are plotting on the wrong year for some reason:
image
     - This may occur because the most recent year is determined based on the daily streamflow data, which is unavailable for this site

- [x] This one (06127900) seems to be having problems too

- [x] Add these statistics to the drop-down statistic group options:
     - Weighted AEP Statistics: 
          - Weighted 80% AEP: 1065
          - Weighted 66.7% AEP: 20
          - Weighted 50% AEP: 513
          - Weighted 42.9% AEP: 512
          - Weighted 20% AEP: 21
          - Weighted 10% AEP: 22
          - Weighted 4% AEP: 23
          - Weighted 2% AEP: 24
          - Weighted 1% AEP: 25
          - Weighted 0.5% AEP: 26
          - Weighted 0.2% AEP: 27
     - Regulated AEP Statistics:
          - Regulated 80% AEP: 1328
          - Regulated 66.7% AEP: 517
          - Regulated 50% AEP: 516
          - Regulated 42.9% AEP: 518
          - Regulated 20% AEP: 519
          - Regulated 10% AEP: 520
          - Regulated 4% AEP: 521
          - Regulated 2% AEP: 522
          - Regulated 1% AEP: 523
          - Regulated 0.5% AEP: 524
          - Regulated 0.2% AEP: 525
- [x] Switch plot to water years (October 1 - September 30)
          - Change peakOnYear dates and mechanism for switching dates with dropdown (choosePeakYear function)
          - Adjust initial load x axis range
          - Change wording for checkbox/select year

### PR DESCRIPTION
**Pull request by maggiejaenicke**
Closes #1707

### CODE REVIEW

File: src/Controllers/GagePageController.ts [hunk 1]
```diff
@@ -2986,6 +3212,84 @@ public chooseFloodStats() {
...
+            });
+        }
+        formattedSelectedPeaks.sort((a, b) => a.x - b.x);
+        formattedEstSelectedPeaks.sort((a, b) => a.x - b.x);
+        chart.series[0].update({data: formattedSelectedPeaks});
+        chart.series[1].update({data: formattedEstSelectedPeaks});
+        chart.yAxis[0].setExtremes();
+        chart.xAxis[0].setExtremes(min, max)
+        chart.series[0].update( {tooltip: {
+            headerFormat:'<b>Annual Peak Streamflow</b><br> Plotted on Water Year  HELLOOOOO: <b>' + this.selectedYear,
```
Comment (harper-wavra) [inline src/Controllers/GagePageController.ts]:
```suggestion
            headerFormat:'<b>Annual Peak Streamflow</b><br> Plotted on Water Year: <b>' + this.selectedYear,
```

File: src/Controllers/GagePageController.ts [hunk 2]
```diff
@@ -889,13 +878,48 @@ module StreamStats.Controllers {
...
+			timeInMillisec -= 15 * 24 * 60 * 60 * 1000; 	// 14 days in milliseconds
+			date.setTime (timeInMillisec);
+            var twoWeeksAgo = new Date(date.getTime() - (date.getTimezoneOffset() * 60000 )) //ending date range two weeks ago to replace with IV
+                    .toISOString()
+                    .split("T")[0];
+            let url;
+            if (this.instFlow.length > 0) {
+                url = 'https://nwis.waterservices.usgs.gov/nwis/dv/?format=json&sites=' + this.gage.code + '&parameterCd=00060&statCd=00003&startDT=1900-01-01&endDT=' + twoWeeksAgo;
+            } else {
+                url = 'https://nwis.waterservices.usgs.gov/nwis/dv/?format=json&sites=' + this.gage.code + '&parameterCd=00060&statCd=00003&startDT=1900-01-01&endDT=2024-01-11';
```
Comment (harper-wavra) [inline src/Controllers/GagePageController.ts]: Should the `endDT` be hard coded here?

**Review by harper-wavra — CHANGES_REQUESTED** [review]
Working well! This was a lot of changes, great job!

- Sometimes the log scale is not an option, [image](https://github.com/USGS-WiM/StreamStats/assets/59063181/f8c4eab8-4e5b-4780-91cf-9a980a3972db). I could not remember if we had talked about this, so maybe this is by design?
- After editing the time period shown on the plot, then clicking 'reset zoom' the page would sometimes freeze for me. This happened the most at gage 1234000. Not sure how easy this will be to fix, so could always make a separate issue.

File: src/Controllers/GagePageController.ts (hunk 1 above)
Comment (maggiejaenicke) [reply src/Controllers/GagePageController.ts]: Lolll

**Comment by maggiejaenicke** [comment]
> * Sometimes the log scale is not an option, [image](https://github.com/USGS-WiM/StreamStats/assets/59063181/f8c4eab8-4e5b-4780-91cf-9a980a3972db). I could not remember if we had talked about this, so maybe this is by design?

Similar to how the heat map determines if it should use log or linear, the log scale check box is only available if the data values do not contain any negative values (which causes the log option not to work on Highcharts). I have it set to not show the checkbox at all in this case (`ng-if="vm.containsNegatives() === false"`), but maybe it would be better/more self-explanatory if it were grayed out and still visible? 


> * After editing the time period shown on the plot, then clicking 'reset zoom' the page would sometimes freeze for me. This happened the most at gage 1234000. Not sure how easy this will be to fix, so could always make a separate issue.

12340000 seems to be super slow for a lot of things, and I think that is because of the huge data gap in the daily value data. The way that gap works is all the values are looped through so that each day has a data row and days without data get a null value. It does have to loop through a lot of data but I don't really know why it's soooo slow. I added some context about this to [this issue](https://github.com/USGS-WiM/StreamStats/issues/1562), so maybe we can work on address loading issues all at once.

**Review by harper-wavra — APPROVED** [review]
> Maybe it would be better/more self-explanatory if it were grayed out and still visible
I like this idea. 

Other than that, things are looking good to me!

**Review by amedenblik — CHANGES_REQUESTED** [review]
Awesome job, Maggie! This is looking sharp.

- [x] On gage 03132000, I am not seeing labels for any of the the horizontal streamgage statistics lines:
image

- As discussed, it would be nice to improve loading times for gages that are especially slow, but we can address this in other issues.

**Comment by maggiejaenicke** [comment]
Notes to self: 

> * [x]  On gage 03132000, I am not seeing labels for any of the the horizontal streamgage statistics lines:

I think I could just set this up in a function similar to `updateShadedStats()` or `updatePeaksAfterZoom()` to update the endpoints of the flood stats lines whenever the axis is resized. Would need this to apply to all of the flood stats, even the ones not currently shown, so that if they are switched in the dropdown, they have the right endpoints.

**Review by amedenblik — APPROVED** [review]

**Comment by amedenblik** [comment]
Deployed to https://dev.streamstats.usgs.gov/gageplots/

**Pull request merged**